shell/edit/resources.cattle.io.backup.vue

```diff
@@ -108,7 +111,14 @@ export default {
     },
 
     encryptionSecretNames() {
-      return this.allSecrets.filter((secret) => (secret.data || {})['encryption-provider-config.yaml'] && secret.metadata.namespace === this.chartNamespace && !secret.metadata?.state?.error).map((secret) => secret.metadata.name);
+      const filtered = this.allSecrets.filter(
+        (secret) => (secret.data || {})['encryption-provider-config.yaml'] &&
+        secret.metadata.namespace === this.chartNamespace &&
+        !secret.metadata?.state?.error &&
+        secret._type === OPAQUE_SECRET_TYPE
+      );
+
+      return filtered.map((secret) => secret.metadata.name).sort();
```

Comment:
Mind that `metadata` is optional and name may not exist for any reason, so it may be worth to filter it further.

Also if you want to chain with more filters I would recommend something like:

```ts
this.allSecrets
      .filter((secret) => (
        secret.data || {})['encryption-provider-config.yaml'] &&
        secret.metadata.namespace === this.chartNamespace &&
        !secret.metadata?.state?.error &&
        secret._type === OPAQUE_SECRET_TYPE
      )
      .map((secret) => secret.metadata.name)
      .sort()
```

Also we should add an unit test for this.